### PR TITLE
Changes based on style and lint checks. Some empty docstrings remain …

### PR DESCRIPTION
**Pull request by coffeedogs**
…to be completed in the future. (final_code_quality_12)

## Repository contributions to the PyBitmessage project

- You can get paid for merged commits if you register at [Tip4Commit](https://tip4commit.com/github/Bitmessage/PyBitmessage)

### Code

- Try to refer to github issue tracker or other permanent sources of discussion about the issue.
- It is clear from the diff *what* you have done, it may be less clear *why* you have done it so explain why this change is necessary rather than what it does

### Documentation

- If there has been a change to the code, there's a good possibility there should be a corresponding change to the documentation 
- If you can't run `fab build_docs` successfully, ask for someone to run it against your branch

### Tests

- If there has been a change to the code, there's a good possibility there should be a corresponding change to the tests
- If you can't run `fab tests` successfully, ask for someone to run it against your branch

## Translations

- For helping with translations, please use [Transifex](https://www.transifex.com/bitmessage-project/pybitmessage/).
- There is no need to submit pull requests for translations.
- For translating technical terms it is recommended to consult the [Microsoft Language Portal](https://www.microsoft.com/Language/en-US/Default.aspx).

### Gitiquette

- Make the pull request against the ["v0.6" branch](https://github.com/Bitmessage/PyBitmessage/tree/v0.6)
- PGP-sign the commits included in the pull request
- Use references to tickets, e.g. `addresses #123` or `fixes #234` in your commit messages
- Try to use a good editor that removes trailing whitespace, highlights potential python issues and uses unix line endings
- If for some reason you don't want to use github, you can submit the patch using Bitmessage to the "bitmessage" chan, or to one of the developers.

### CODE REVIEW

File: src/addresses.py [hunk 1]
```diff
@@ -59,6 +67,7 @@ def decodeBase58(string, alphabet=ALPHABET):
 
 
 def encodeVarint(integer):
+    """"""
```
Comment (PeterSurda) [inline src/addresses.py]:
```suggestion
    """Convert integer into varint bytes"""
```

File: src/addresses.py [hunk 2]
```diff
@@ -76,6 +85,7 @@ def encodeVarint(integer):
 
 
 class varintDecodeError(Exception):
+    """"""
```
Comment (PeterSurda) [inline src/addresses.py]:
```suggestion
    """Exception class for decoding varint data"""
```

File: src/addresses.py [hunk 3]
```diff
@@ -135,6 +145,7 @@ def decodeVarint(data):
 
 
 def calculateInventoryHash(data):
+    """"""
```
Comment (PeterSurda) [inline src/addresses.py]:
```suggestion
    """Calculate inventory hash from object data"""
```

File: src/addresses.py [hunk 4]
```diff
@@ -143,6 +154,7 @@ def calculateInventoryHash(data):
 
 
 def encodeAddress(version, stream, ripe):
+    """"""
```
Comment (PeterSurda) [inline src/addresses.py]:
```suggestion
    """Convert ripe to address"""
```

File: src/addresses.py [hunk 5]
```diff
@@ -271,13 +277,13 @@ def decodeAddress(address):
...
-            x00string = '\x00' * (20 - len(embeddedRipeData))
-            return status, addressVersionNumber, streamNumber, \
-                x00string + embeddedRipeData
+        x00string = '\x00' * (20 - len(embeddedRipeData))
+        return status, addressVersionNumber, streamNumber, \
+            x00string + embeddedRipeData
 
 
 def addBMIfNotPresent(address):
+    """"""
```
Comment (PeterSurda) [inline src/addresses.py]:
```suggestion
    """Prepend BM- to an address if it doesn't already have it"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 6]
```diff
@@ -1,29 +1,39 @@
...
+import widgets
+from account import AccountMixin, GatewayAccount, MailchuckAccount, accountClass, getSortedAccounts
+from addresses import addBMIfNotPresent, decodeAddress, encodeVarint
 from inventory import Inventory
+from retranslateui import RetranslateMixin
+from tr import _translate
 
 
 class AddressCheckMixin(object):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """Base address validation class for QT UI"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 7]
```diff
@@ -1,29 +1,39 @@
...
+        QtCore.QObject.connect(  # pylint: disable=no-member
+            self.lineEditAddress,
+            QtCore.SIGNAL("textChanged(QString)"),
+            self.addressChanged)
 
     def _onSuccess(self, addressVersion, streamNumber, ripe):
         pass
 
     def addressChanged(self, QString):
+        """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
        """Address validation callback, performs validation and gives feedback"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 8]
```diff
@@ -71,11 +81,14 @@ def addressChanged(self, QString):
 
 
 class AddressDataDialog(QtGui.QDialog, AddressCheckMixin):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """QDialog with Bitmessage address validation"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 9]
```diff
@@ -71,11 +81,14 @@ def addressChanged(self, QString):
...
+
     def __init__(self, parent):
         super(AddressDataDialog, self).__init__(parent)
         self.parent = parent
 
     def accept(self):
+        """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
        """Callback for QDIalog accepting value"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 10]
```diff
@@ -90,6 +103,7 @@ def accept(self):
 
 
 class AddAddressDialog(AddressDataDialog, RetranslateMixin):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """QDialog for adding a new address, with validation and translation"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 11]
```diff
@@ -100,6 +114,7 @@ def __init__(self, parent=None, address=None):
 
 
 class NewAddressDialog(QtGui.QDialog, RetranslateMixin):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """QDialog for generating a new address, with translation"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 12]
```diff
@@ -115,6 +130,7 @@ def __init__(self, parent=None):
         self.show()
 
     def accept(self):
+        """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
        """accept callback"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 13]
```diff
@@ -160,6 +176,7 @@ def accept(self):
 
 
 class NewSubscriptionDialog(AddressDataDialog, RetranslateMixin):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """QDialog for subscribing to an address, with validation and translation"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 14]
```diff
@@ -202,6 +219,7 @@ def _onSuccess(self, addressVersion, streamNumber, ripe):
 
 
 class RegenerateAddressesDialog(QtGui.QDialog, RetranslateMixin):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """QDialog for regenerating deterministic addresses, with translation"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 15]
```diff
@@ -210,6 +228,7 @@ def __init__(self, parent=None):
 
 
 class SpecialAddressBehaviorDialog(QtGui.QDialog, RetranslateMixin):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """QDialog for special address behaviour (e.g. mailing list functionality), with translation"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 16]
```diff
@@ -250,6 +269,7 @@ def __init__(self, parent=None, config=None):
         self.show()
 
     def accept(self):
+        """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
        """Accept callback"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 17]
```diff
@@ -267,14 +287,15 @@ def accept(self):
             self.config.set(str(self.address), 'mailinglistname', str(
                 self.lineEditMailingListName.text().toUtf8()))
             self.parent.setCurrentItemColor(
-                QtGui.QColor(137, 04, 177))  # magenta
+                QtGui.QColor(137, 0o4, 177))  # magenta
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]: just 4

File: src/bitmessageqt/address_dialogs.py [hunk 18]
```diff
@@ -267,14 +287,15 @@ def accept(self):
...
         self.parent.rerenderComboBoxSendFrom()
         self.parent.rerenderComboBoxSendFromBroadcast()
         self.config.save()
         self.parent.rerenderMessagelistToLabels()
 
 
 class EmailGatewayDialog(QtGui.QDialog, RetranslateMixin):
+    """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
    """QDialog for email gateway control, with translation"""
```

File: src/bitmessageqt/address_dialogs.py [hunk 19]
```diff
@@ -315,6 +336,7 @@ def __init__(self, parent, config=None, account=None):
         QtGui.QWidget.resize(self, QtGui.QWidget.sizeHint(self))
 
     def accept(self):
+        """"""
```
Comment (PeterSurda) [inline src/bitmessageqt/address_dialogs.py]:
```suggestion
        """Accept callback"""
```

File: src/class_smtpDeliver.py [hunk 20]
```diff
@@ -1,52 +1,59 @@
...
 import state
+from bmconfigparser import BMConfigParser
+from debug import logger
+from helper_threading import StoppableThread
 
 SMTPDOMAIN = "bmaddr.lan"
 
+
 class smtpDeliver(threading.Thread, StoppableThread):
+    """"""
```
Comment (PeterSurda) [inline src/class_smtpDeliver.py]:
```suggestion
    """SMTP client thread for delivery"""
```

File: src/class_smtpDeliver.py [hunk 21]
```diff
@@ -1,52 +1,59 @@
...
         try:
-            queues.UISignallerQueue.put(("stopThread", "data"))
+            queues.UISignallerQueue.put(("stopThread", "data"))  # pylint: disable=no-member
         except:
             pass
         super(smtpDeliver, self).stopThread()
 
     @classmethod
     def get(cls):
+        """"""
```
Comment (PeterSurda) [inline src/class_smtpDeliver.py]:
```suggestion
        """(probably) Singleton functionality"""
```

File: src/randomtrackingdict.py [hunk 22]
```diff
@@ -1,12 +1,22 @@
...
+
 import random
 from threading import RLock
 from time import time
+
 import helper_random
 
+
 class RandomTrackingDict(object):
+    """"""
```
Comment (PeterSurda) [inline src/randomtrackingdict.py]:
```suggestion
    """Dict with randomised order and tracking. Keeps a track of how many items have been requested from the dict, and timeouts. Resets after all objects have been retrieved and timed out. The main purpose of this isn't as much putting related code together as performance optimisation and anonymisation of downloading of objects from other peers. If done using a standard dict or array, it takes too much CPU (and looks convoluted). Randomisation helps with anonymity"""
```

File: src/randomtrackingdict.py [hunk 23]
```diff
@@ -67,20 +77,25 @@ def __delitem__(self, key):
             self.len -= 1
 
     def setMaxPending(self, maxPending):
+        """"""
```
Comment (PeterSurda) [inline src/randomtrackingdict.py]:
```suggestion
        """Sets maximum number of objects that can be retrieved from the class simultaneously as long as there is no timeout"""
```

File: src/randomtrackingdict.py [hunk 24]
```diff
@@ -67,20 +77,25 @@ def __delitem__(self, key):
...
         self.maxPending = maxPending
 
     def setPendingTimeout(self, pendingTimeout):
+        """"""
```
Comment (PeterSurda) [inline src/randomtrackingdict.py]:
```suggestion
        """Sets how long to wait for a timeout if max pending is reached (or all objects have been retrived)"""
```

File: src/randomtrackingdict.py [hunk 25]
```diff
@@ -67,20 +77,25 @@ def __delitem__(self, key):
...
         self.pendingTimeout = pendingTimeout
 
     def setLastObject(self):
         """Update timestamp for tracking of received objects"""
         self.lastObject = time()
 
     def randomKeys(self, count=1):
+        """"""
```
Comment (PeterSurda) [inline src/randomtrackingdict.py]:
```suggestion
        """Retrieve count random keys from the dict that haven't already been retrieved"""
```

File: src/randomtrackingdict.py [hunk 26]
```diff
@@ -99,43 +114,40 @@ def randomKeys(self, count=1):
             self.lastPoll = time()
             return retval
 
+
 if __name__ == '__main__':
+
+    # pylint: disable=redefined-outer-name
     def randString():
+        """"""
```
Comment (PeterSurda) [inline src/randomtrackingdict.py]:
```suggestion
        """helper function for tests, generates a random string"""
```

**Comment by coffeedogs** [comment]
Not sure why I deleted src/api_client.py but I've restored it and moved it to /dev

**Comment by PeterSurda** [comment]
Check don't handle renaming of files, creating false negatives. I'll work on fixing it.

**Comment by PeterSurda** [comment]
Renames are now handled correctly.

**Pull request merged**